### DISCUSSION TITLE
ANS proposition to include an ASA Stats IFrame in the new .algo profile they're planning to launch

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#98] [General]
By SaiM in [Discord](https://discord.com/channels/906917846754418770/910524302531649566/978327121195827200):

- - - - -

Hi all, this is Sai from ANS. We love ASA Stats and (with your permission) we hope to include an ASA Stats IFrame in the new .algo profile we're planning to launch. Imagine users entering something like sai.algo.website on their browser which will display general domain info, NFT collection from Rand Gallery, and Asset profile from Asastats of the wallet address. 
Looking forward to hear y'alls thoughts on this, happy to discuss more and answer questions

- - - - -

**Comment by bear1bear2bear3** [comment]
Sounds pretty darn awesome to me!! What steps are needed for this to work out?

**Reply by Babbexx22** [reply]
It's a good proposition. Let's have the discussion pushed forward.. And also the steps as raised by Bear1Bear2Bear3

**Reply by smedury** [reply]
We are thrilled to hear that. To reiterate, ASA Stats provides a unique overview and set of insights into an Algorand wallet's asset holdings and we think the .algo.website would be a great opportunity for Algorand users to display their portfolios alongside NFT collections. We're excited :)

We just need the ASA Stats website to enable the option to allow being imported as IFrame (by *.algo.website domain). Lalith will respond with more specifics. Here's a helpful StackOverflow [answer]( https://stackoverflow.com/questions/39483348/how-to-allow-iframe-embedding-only-for-whitelisted-websites).

**Reply by ipaleka** [reply]
Hi @smedury , thanks for joining the discussion here. :)

Do you have any approximations for the launch date?

**Reply by smedury** [reply]
Hi @ipaleka, we're close to the finish line but we haven't set a launch date yet. We're thinking in a couple of days maybe, depending on how long it might take to enable this on ASA Stats website. Everything else is ready on our end except the ASAStats IFrame. But no rush, take your time. I understand ASAStats may already have a lot on plate.

**Reply by ipaleka** [reply]
> Hi @ipaleka, we're close to the finish line but we haven't set a launch date yet. We're thinking in a couple of days maybe, depending on how long it might take to enable this on ASA Stats website. Everything else is ready on our end except the ASAStats IFrame. But no rush, take your time. I understand ASAStats may already have a lot on plate.

Oh, that's not the reason why I asked that, it's just an entry in our revers!e proxy configuration to allow you that.

We'd be encouraging and even sponsoring mobile apps and even other websites to use our API and provide such a functionality in much more customizable way than iFrame, so my question has been directed that way.

But that's not an option obviously as we are still in discussion phase for our API.

So, we'll wait another day for some other members to eventually bring their opinions about this.

**Comment by wxivd** [comment]
I think this is beneficial for ASA Stats and ANS on many levels.

1. Marketing advantage: 
- People searching domains would get to explore wallet contents, this could be ideal for NFT projects and ASA projects.
- More people get to try our portfolio tracker. I like to think retention rate for users who try it for the first time is high. 


2. Collaboration with the view of creating a future business agreement, Win-win scenario
- ANS utilises our service to serve their user-base.
- Get ANS feedback regarding some widgets they would Need us to develop or they to sponsor. This helps us with other naming services as well. 

I suggest the first 3 months free of charge, giving ANS and ASA Stats an opportunity to try things out. 

In the meanwhile we get to think of and find a win-win business agreement. The time frame is suitable to
- Measure interaction and user base. 
- Analyse number of clicks on ANS website.
- Tailoring pricing packages, ASA Stats selling ANS our services (e.g. number of profile search per month. 3 pricing packages or something).

**Reply by smedury** [reply]
I like that idea, three months is the perfect timeframe. We're also launching ANS DAO smart contracts and infrastructure, and we envision the DAO supporting development in ANS and community projects such as ASA Stats so supporting the development of widgets and other software components to enhance engagement is aligned with the core vision.

I have to say, we're a little skeptical about pay per click or pay per search because that doesn't meet the name service community-goods operating model of ANS. Perhaps we can explore alternate arrangements like the ANS team can collaborate and contribute to ASA Stats DAO smart contracts development or share ANS DAO infrastructure like voting GUI or source code etc.

**Comment by smedury** [comment]
@ipaleka -- Thanks for enabling IFrames on algo.website.

ASA Stats team and community, 

We're ready with the updates to algo.website project that includes Asset Portfolio from ASA Stats. Give it a try by just adding .website to .algo name in your favorite browser. Please note, this is still in the Beta phase. We would love to hear your thoughts and suggestions for improvement.

[ans.algo.website](https://ans.algo.website)        [sai.algo.website](https://sai.algo.website)          [rand.algo.website](https://rand.algo.website)

Further, we plan to work towards testing and integrating ASA Stats SDK when it's ready. IFrames are not ideal and we can enhance user experience much more by using SDK.

Thanks,
Sai

**Reply by AlgoRhythMatic** [reply]
> IFrames are not ideal and we can enhance user experience much more by using SDK.

This is a nice proof of concept though, and is exactly the type of partnership we are looking for to build up our communities.  Thank you for participating in our conversation, and for bringing a result so quickly!
